stdlib/public/Concurrency/Task.cpp

```diff
@@ -63,10 +63,12 @@ FutureFragment::Status AsyncTask::waitFuture(AsyncTask *waitingTask) {
     switch (queueHead.getStatus()) {
     case Status::Error:
     case Status::Success:
+      _swift_tsan_acquire(static_cast<Job *>(this));
       // The task is done; we don't need to wait.
       return queueHead.getStatus();
 
     case Status::Executing:
+      _swift_tsan_release(static_cast<Job *>(waitingTask));
```

Comment:
One more implementation note: at this point, we actually can't guarantee yet that we are going to execute the `while (true)` loop with the `Executing` case.  The `if (... waitQueue.compare_exchange_weak ...)` below could fail and when we query the status again it could have changed.  So we are potentially adding a superfluous edge.  However, I don't think we want to put this call inside the `if` below.  This would mean that we put ourselves on the "wrong side of the race", i.e., when we loose this race we could produce a false positive (in very rare situations).